clinica/pydra/t1_volume/t1_volume_utils.py

```diff
@@ -0,0 +1,536 @@
+            if save_warped_modulated:
+                warped_modulated = True
+
+        if i in dartel_tissues:
+            dartel_input = True
+
+        tissues.append(
+            (
+                (tissue_map, i),
+                n_gaussians,
+                (native_space, dartel_input),
+                (warped_unmodulated, warped_modulated),
+            )
+        )
+
+    return tissues
+
+
+def is_centered(nii_volume: PathLike, threshold_l2: int = 50) -> bool:
+    """Checks if a NIfTI volume is centered on the origin of the world coordinate system.
+
+    Parameters
+    ---------
+    nii_volume : PathLike
+        path to NIfTI volume
+    threshold_l2: int
+        Maximum distance between origin of the world coordinate system and the center of the volume to
+        be considered centered. The threshold were SPM segmentation stops working is around 100 mm (it was determined empirically after several
+        trials on a generated dataset), so default value is 50mm in order to have a security margin, even when dealing with co-registered files afterward.
+
+    Returns
+    -------
+        bool
+
+    Notes
+    ------
+
+    SPM has troubles to segment files if the center of the volume is not close from the origin of the world coordinate
+    system. A series of experiment have been conducted: we take a volume whose center is on the origin of the world
+    coordinate system. We add an offset using coordinates of affine matrix [0, 3], [1, 3], [2, 3] (or by modifying the
+    header['srow_x'][3], header['srow_y'][3], header['srow_z'][3], this is strictly equivalent).
+
+    It has been determined that volumes were still segmented with SPM when the L2 distance between origin and center of
+    the volume did not exceed 100 mm. Above this distance, either the volume is either not segmented (SPM error), or the
+    produced segmentation is wrong (not the shape of a brain anymore)
+    """
+
+    import numpy as np
+
+    center = get_world_coordinate_of_center(nii_volume)
+
+    distance_from_origin = np.linalg.norm(center, ord=2)
+
+    if distance_from_origin < threshold_l2:
+        return True
+    else:
+        # If center is a np.nan,
+        return False
+
+
+def get_world_coordinate_of_center(nii_volume: PathLike) -> ndarray:
+    """Extract the world coordinates of the center of the image.
+
+    Parameters
+    ---------
+    nii_volume : PathLike
+        path to nii volume
+
+    Returns
+    -------
+    tuple
+        coordinates in the world space
+
+    References
+    ------
+    https://brainder.org/2012/09/23/the-nifti-file-format/
+
+    """
+
+    from os.path import isfile
+
+    import nibabel as nib
+    import numpy as np
+    from nibabel.filebasedimages import ImageFileError
+
+    # from clinica.utils.stream import cprint
+    # assert isinstance(nii_volume, str), "input argument nii_volume must be a str"
+
+    assert isfile(nii_volume), "input argument must be a path to a file"
```

Comment:
I would raise a proper error instead of using an assertion here...

Reply:
I'm not sure I agree, it seems at this point `nii_volume` has been used in other functions (the input comes from another function call). Granted, in the current use case it has never changed but if we are to use an exception it would rather be in the function that first receives the nifti list.

The assert here seems to function more as a sanity check to verity that our previous functions have not modified this input's type.

Reply:
The issue with an error here is that we will be notifying the user about a problem with an input that he would have not provided himself. I'm not sure I understand the purpose of that?

> If that's indeed the underlying reason, then this is very scary. Functions that modifies their inputs are a red flag.
Exactly. It would be extremely scary and hence raising an error seem appropriate? 

I'll leave it for now and we can discuss in person if necessary.

Reply:
> The issue with an error here is that we will be notifying the user about a problem with an input that he would have not provided himself. I'm not sure I understand the purpose of that?

I believe Nicolas and I are talking in the more general sense of error handling. Take you custom implementation of a divison function and the case where the denominator is `0`. You may decide to `assert`, on it or raise a `ZeroDivisionError` like Python does in its standard implementation. The latter provides greater freedom to implement your own error handling strategy (even a YOLO mode `try / except: pass`) as opposed to the former.

Now back to our use case, there is nothing that prevents us for getting the same behaviour of the current `assert` implementation with a `try / catch` block.

> Exactly. It would be extremely scary and hence crashing seems appropriate, no(?)

Ensuring we are not using functions that modify their inputs would be more of an improvement. Otherwise, we are just reproducing bad habits from the legacy Clinica code of doing defensive programming and shotgun parsing everywhere.

I am not willing to argue on this further to be honest.

Reply:
I'm also talking about the more general sense of error handling. 

With the example you provided the distinction I'm making is from whereby the `0`comes. If it is a user input then yes, error handling make sense as you raise a proper exception that the user can then address. On the other hand, if the value you are checking comes from inner processing, then the assert is here as a failsafe for the code not to continue.
That being said, I have no strong opinions against handling this with an error, I was just pointing out to the fact that the error message (saying the variabel is not a nifti file) will not be informative to the user at this point in the code, that is all. Will change.

On the point of legacy code: legacy code is not the scope of this PR and this PR was solely renaming and reusing utility functions.The flaws directly in the legacy code can and should be addressed if properly planned for. 

On a side note when I said "I'll leave it for now and we can discuss in person if necessary." it was to avoid a further reply in here.